### PR TITLE
chore: Migrate draft CMOs to hearing orders drafts bundle (FPLA-2710)

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-2710

### Change description ###
Migrate draft CMOs to hearing orders bundles

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by aliveni** [comment]
Tested data migration in local environment:
image

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 1]
```diff
@@ -46,10 +74,84 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             caseDetails.getData().put("hiddenOrders", hiddenOrders);
         }
 
+        if ("FPLA-2710".equals(migrationId)) {
+            CaseData caseData = getCaseData(caseDetails);
+            if (casesWithUploadDraftCMOs.contains(caseData.getId()) && isNotEmpty(caseData.getDraftUploadedCMOs())) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:79]: i would check if caseData.getHearingOrdersBundlesDrafts() is not empty. If there is new structure then i guess data was already migrated

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 2]
```diff
@@ -46,10 +74,84 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
...
+                    "Completed migration to Hearing orders draft bundles. case reference {} Number of bundles {}",
+                    caseData.getId(), migratedBundles.size());
+            }
+        }
+
         caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
 
+    private List<Element<HearingOrdersBundle>> migrateDraftCMOsToHearingBundles(CaseData caseData) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:98]: could use maybe DraftOrderService.migrateCmoDraftToOrdersBundles

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 3]
```diff
@@ -46,10 +74,84 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
...
+
+        draftUploadedCMOs.forEach(draftElement -> {
+            Optional<Element<HearingBooking>> hearingOptional = caseData.getHearingLinkedToCMO(draftElement.getId());
+
+            // if hearing exists linked with draft CMO
+            hearingOptional.ifPresent(hearingElement -> {
+                String title = draftElement.getValue().getType() == AGREED_CMO ? "Agreed CMO discussed at hearing"
+                    : "Draft CMO from advocates' meeting";
+
+                draftElement.getValue().toBuilder()
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:117]: I don't think this is updating the title -> tobuilder creates another instance that is not reassigned

Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java]: This code is replaced with `DraftOrderService. migrateCmoDraftToOrdersBundles`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java (hunk 1 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java]: updated

**Review by tomaszpowroznik — DISMISSED** [review]

**Review by tomaszpowroznik — APPROVED** [review]

**Pull request merged**